iota/commands/extended/traverse_bundle.py

```diff
@@ -55,10 +55,13 @@ def _traverse_bundle(self, txn_hash, target_bundle_hash):
             GetTrytesCommand(self.adapter)(hashes=[txn_hash])['trytes']
         )  # type: List[TryteString]
 
-        if not trytes:
+        # If no tx was found by the node for txn_hash, it returns 9s,
+        # so we check here if it returned all 9s trytes.
+        if not trytes or trytes == [TransactionTrytes('')]:
             raise with_context(
                 exc=BadApiResponse(
-                    'Bundle transactions not visible '
+                    'Could not get trytes of bundle transaction from the Tangle. '
+                    'Bundle transactions not visible.'
```

Comment:
💭 Wondering if this also indicates an issue with the IRI node — should `getTrytes` ever return an all-9's value?

Reply:
According to the [IRI API reference](https://docs.iota.org/docs/node-software/0.1/iri/references/api-reference#gettrytes), it may return string of 9s when the node doesn't have the transaction in its ledger.

Reply:
Ah, I see.   That makes sense, since the client can pass multiple hashes to `getTrytes`.